model-optimizer/extensions/ops/gatherelements.py

```diff
@@ -0,0 +1,71 @@
+"""
+ Copyright (C) 2017-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from mo.graph.graph import Node, Graph
+from mo.ops.op import Op, PermuteAttrs
+
+
+class GatherElements(Op):
+    op = 'GatherElements'
+
+    def __init__(self, graph: Graph, attrs: dict):
+        super().__init__(graph, {
+            'op': self.op,
+            'type': self.op,
+            'version': 'opset6',
+            'infer': self.infer,
+            'in_ports_count': 2,
+            'out_ports_count': 1,
+            'axis': 0,
+        }, attrs)
+
+    def backend_attrs(self):
+        return ['axis']
+
+    @staticmethod
+    def infer(node: Node):
+        data_shape = node.in_port(0).data.get_shape()
+        indices_shape = node.in_port(1).data.get_shape()
+        axis = node.axis
+        data_rank = len(data_shape)
+
+        assert data_rank == len(indices_shape), f'data and indices inputs for node {node.name} must be of the ' \
+                                                f'same rank. Got {data_rank} and {len(indices_shape)}'
+        assert -data_rank < axis < data_rank, f'axis for node {node.name} must be within interval ' \
+                                              f'[{-data_rank},  {data_rank - 1}]. Instead axis={axis}'
```

Comment:
Since py3.5 is no longer supported by python community and in OpenVINO distribution do not have python api binding for py version < 3.6 I decided to use f-strings.

Reply:
Unfortunately, it is not enough to start using py3.6+ features. Please, return to a regular format string

Reply:
Agreed. Returned back `.format(`